### PR TITLE
Create users in the A3 backend as well as in A3

### PR DESCRIPTION
**Pull request by ChristianSi**

### CODE REVIEW

File: src/Thentos/Backend/Core.hs [hunk 1]
```diff
@@ -101,6 +101,8 @@ actionErrorToServantErr e = do
         { errBody = "error accessing user info" }
     _thentos (SsoErrorCouldNotGetAccessToken _) = pure $ err500
         { errBody = "error retrieving access token" }
+    _thentos (A3BackendError msg) = pure $ err500
+        { errBody = "error talking to A3 backend: " <> cs msg }
```
Comment (fisx) [inline src/Thentos/Backend/Core.hs]: I just pushed 6533ee7 regarding this.

A3BackendError is very untyped.  There is no strong reason to not do that besides that it's not necessary and not idiomatic.  I would use different types `A3BackendBadResponseCode`, `A3BackendNoSuchUser`, ..., and make each carry the details in typed form.  See 6533ee7 for more explanation.

File: src/Thentos/Transaction/Transactions.hs [hunk 2]
```diff
@@ -49,6 +49,10 @@ assertUser mUid user = polyQuery $ ask >>= \ db ->
        | userFacetExists (^. userEmail) mUid user db -> throwT UserEmailAlreadyExists
        | True -> return ()
 
+-- | Assert that no user with the same name or email address already exists in the DB.
+trans_assertUserIsNew :: (AsDB db) => User -> ThentosQuery db ()
+trans_assertUserIsNew = assertUser Nothing
```
Comment (fisx) [inline src/Thentos/Transaction/Transactions.hs]: see earlier comment on the missing exception in `addUnconfirmedUserWithId`.  if you fix that, this here may become unnecessary.  in that case i'd like to keep the api simple.  perhaps `lookupUser` can cover for this one?

File: src/Thentos/Transaction/Transactions.hs [hunk 3]
```diff
@@ -530,7 +543,8 @@ $(deriveSafeCopy 0 'base ''UpdateUserFieldOp)
 
 transaction_names :: [Name]
 transaction_names =
-    [ 'trans_allUserIds
```
Comment (fisx) [inline src/Thentos/Transaction/Transactions.hs]: thank you.  (-:  i've been meaning to do that for a while now.

File: src/Thentos/Transaction/Transactions.hs (hunk 2 above)
Comment (fisx) [inline src/Thentos/Transaction/Transactions.hs]: @ChristianSi has pointed out that this has nothing to do with `UserId`, but checks freshness of name and email.  Please ignore previous comment.

**Comment by ChristianSi** [comment]
As a side effect, my latest commit fixes #190.

**Comment by ChristianSi** [comment]
@fisx I think it makes sense to merge this now so as not to interfere with the big refactoring. Some of your comments are still unaddressed, but I'll take care of them in a different PR.

**Comment by fisx** [comment]
looks good.  i will unassign me; @ChristianSi, if you find that travis is happy, you can merge.  if you don't, i'll do it tonight.

**Pull request merged**